csi/volumeutils.py

```diff
@@ -1124,6 +1124,54 @@ def mount_glusterfs(volume, mountpoint, storage_options="", is_client=False):
     else:
         volname = volume["name"]
 
+    if volume['type'] == 'External':
+        # Try to mount the Host Volume, handle failure if
+        # already mounted
+        with mount_lock:
+            mount_glusterfs_with_host(volname,
+                                      mountpoint,
+                                      volume['g_host'],
+                                      volume['g_options'],
+                                      is_client)
+        use_gluster_quota = False
+        if (os.path.isfile("/etc/secret-volume/ssh-privatekey")
+            and "SECRET_GLUSTERQUOTA_SSH_USERNAME" in os.environ):
+            use_gluster_quota = True
+        secret_private_key = "/etc/secret-volume/ssh-privatekey"
+        secret_username = os.environ.get('SECRET_GLUSTERQUOTA_SSH_USERNAME', None)
+
+        # SSH into only first reachable host in volume['g_host'] entry
+        g_host = reachable_host(volume['g_host'])
+
+        if g_host is None:
+            logging.error(logf("All hosts are not reachable"))
+            return
+
+        if use_gluster_quota is False:
+            logging.debug(logf("Do not set quota-deem-statfs"))
+        else:
+            logging.debug(logf("Set quota-deem-statfs for gluster directory Quota"))
+            quota_deem_cmd = [
+                "ssh",
+                "-oStrictHostKeyChecking=no",
+                "-i",
+                "%s" % secret_private_key,
+                "%s@%s" % (secret_username, g_host),
+                "sudo",
+                "gluster",
+                "volume",
+                "set",
+                "%s" % volume['g_volname'],
+                "quota-deem-statfs",
+                "on"
+            ]
+            try:
+                execute(*quota_deem_cmd)
+            except CommandException as err:
+                errmsg = "Unable to set quota-deem-statfs via ssh"
+                logging.error(logf(errmsg, error=err))
+                raise err
+        return mountpoint
 
     generate_client_volfile(volname)
```

Comment:
- as `mount_glusterfs` is refactored, changes to `generate_client_volfile` will not have any effect now
- so, we shouldn't be tolerating absence of `bricks` in kadalu-info cm and it should fail to generate client volfile

Reply:
1. Yes, except for External there should be bricks field in CR.
2. For external we delegate pulling volfile from external gluster to gluster binary itself
3. Ack, validating bricks field in `generating_client_volfile` seems sensible and make it return false on failure and stop proceeding with mounting volume/glusterfs.